### PR TITLE
Refactor AWS auth across plugins and upgrade SDK

### PR DESCRIPTION
**Pull request by manojlds**

### CODE REVIEW

**Comment by manojlds** [comment]
@ashwanthkumar Let me know if this is inline with what you had in mind in #41

File: utils/src/main/java/com/indix/gocd/utils/store/S3ArtifactStore.java [hunk 1]
```diff
@@ -187,4 +199,13 @@ public RevisionStatus getLatest(AmazonS3Client client, Artifact artifact) {
         }
         return null;
     }
+
+    public static AmazonS3 getS3client(GoEnvironment env) {
+        AmazonS3ClientBuilder amazonS3ClientBuilder = AmazonS3ClientBuilder.standard();
```
Comment (ashwanthkumar) [inline utils/src/main/java/com/indix/gocd/utils/store/S3ArtifactStore.java]: @manojlds I'm not aware of the latest client authentication chain. But does it cover all the things as [specified here](https://github.com/indix/gocd-s3-artifacts/pull/41/files#diff-e36fde5bc02a23e18c6b99a90ab85c63R40)?

Comment (manojlds) [inline utils/src/main/java/com/indix/gocd/utils/store/S3ArtifactStore.java]: The below in the code you referenced:

`new EnvironmentVariableCredentialsProvider()` - Present
`new SystemPropertiesCredentialsProvider()` - Present
`new ProfileCredentialsProvider()` - Present
`new InstanceProfileCredentialsProvider()` - Present
`new BasicAWSCredentialsProvider(accessKey, secretKey)` - Not needed. Not that the code gets the creds from env vars and then passes as params.
`new AnonAWSCredentialsProvider()` - Present

Also adds ECS support :)

Comment (ashwanthkumar) [inline utils/src/main/java/com/indix/gocd/utils/store/S3ArtifactStore.java]: 👍  merge away!

**Pull request merged**